Add GitHub codespaces (#9)
irachle/JavaInstrumentationWorkshop: DEV_ENV_SETUP.md

***

> Setup you dev environment.

Reviewer: ```suggestion
Set up your development environment.
```

Author: Thanks. I've changed it.

---------------------------
If you are developing in Java and have maven installed, you can skip this setup and work with the work environment of your choice.

> However if you don't have an active Java development environment, please follow the following to make sure you are all set up and ready to write the code for the excercise.

Reviewer: ```suggestion
However, if you don't have an active Java development environment, please follow these instructions to ensure you are all set up and ready to write the code for the exercise.
```

Author: fixed

// todo - the purpese of this readme is to help setup a working environment in conferences where participants have almost no knowledge in Java development. 
// todo - this is what we thoght might help in conferences like Hackeriot where the level of Java knowlegde is CS graduate level at most and bootcamp or personal interest knowledge is even more common.
// todo - pass all text with chatgpt.
// todo - write this as a trubleshoot guide - with how to test that the step was success full and perhaps guide how to fix common problems (path, restart, and others) 

> Prepare work environment with Maven and IDE 

Reviewer: ```suggestion
Prepare the work environment with Maven and an IDE 
```

--------------------------------------------

> To prepare your work environment, follow these steps:

Reviewer: I'd drop this sentence - it's obvious from the heading

1. IDE. \

>    If you don't have a preferred working IDE, download and install [Intellij](https://www.jetbrains.com/idea/download/) Community Edition for free. Make sure you download the Community Edition.

Reviewer: The second sentence is superfelous - you already said download the Community Edition

2. JDK \
   If you don't have a JDK installed, download any version of JDK. The project was tested with JDK 1.8 and JDK 23. You can use IntelliJ to download the JDK by following the steps explained [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).
   // todo - write how to test that the JDK is installed by running java --version (can also wok for JRE), how to set JAVA_HOME, perhaps suggest to download using intellij but give the option to download independently. jdk23 will work 
4. Maven \
   Download Maven, the build tool used in this project, from [here](https://maven.apache.org/download.cgi) and extract it to a folder (no installation required). 
   Then, add the maven/build folder to the PATH variable.

>    // todo - write here how to make sure maven is installed. 

Reviewer: This is a workshop for juniors - I'd also add the exact steps to add it to the path, for both Windows and macOS.

Author: I have changed this part and suggested to work with the Maven that is bundled with Intellij (or even better to work in the codespaces environment)
